### ISSUE TITLE
README is backwards

### ISSUE THREAD

**Issue by skade** [serde-rs/serde#161]
After using `serde` for a while, I'm now pretty sure that the README is backwards. It has:

1. How to use serdes convenience features for nightly
2. How to use them in a hacky way on stable
3. How does serde work?

The correct order should be:

1. How does serde work?
2. How to use serdes convenience features for nightly
3. How to use them on stable if you absolutely must

Currently, the README is very confusing for people that are not deep into the language and don't know about feature flags and the difference between nightly and stable, but would be okay with some boilerplate. It also makes the impression that the framework is not fully usable on stable.

**Comment by erickt** [comment]
That's a great point! I'll shuffle the order.

**Comment by derekdreery** [comment]
I agree. It's great once you get the hang of it - but I struggled.

My example was I had  an object like
```rust
Result<T, Err> where T: serde::Serialize
```
and I wanted it to become
```json
{"success": true, "data": "test"}
```
for example if the result was `Ok(String::from("test"))`.

An example like this would be great - I can paste it here if you think it would be helpful

**Comment by JohnHeitmann** [comment]
I think the build system emphasis should be reversed from skade@'s suggestion. Behind the scenes stable+syntex is hacky, however the user-facing surface area only has a few minor warts. Since Rust users in general are encouraged to stick with Rust stable, and since serde's stable integration is pretty workable, I think the docs should read like, "Here's the one true way to do it (stable+syntex). And oh, by the way check out the appendix for ways to smooth it over in nightly."

The dithering between stable and nightly in the getting started docs adds a needless layer of complication for newcomers. It'd be nice to smooth that over and give new users one less choice to think about.

I'd volunteer for an update pass on the docs and starter examples if this approach gained consensus.

**Comment by skade** [comment]
> (stable+syntex)

I'm not sure having this is the one true way. I avoid syntex for that use and really don't like serdes serialization macros.

**Comment by dtolnay** [comment]
Erick addressed the order in #428. Please reopen if you know a better order.
